src/web/app/pages-instructor/instructor-student-records-page/instructor-student-records-page.component.spec.ts

```diff
@@ -50,4 +50,13 @@ describe('InstructorStudentRecordsPageComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should output a correctly formatted url string', () => {
+    expect(component.photoUrl)
+      .toEqual('http://localhost:8080/webapi/student/profilePic?courseid=undefined&studentemail=undefined');
```

Comment:
Great! You are on the right track. The only thing left is to inject real profile into the component. You can search on existing code (e.g. other test files) or take a look at `student-course-details-page.component.spec.ts` to get an idea of how to assign value to component during testing.

Reply:
Sorry, perhaps I should have phrased my question a little better. I'm not sure how to mock the queryParams of the `ngOnInit` callback.
```
this.route.queryParams.subscribe((queryParams: any) => {
      this.courseId = queryParams.courseid;
      this.studentEmail = queryParams.studentemail;

      this.loadStudentRecords();
      this.photoUrl
          = `${environment.backendUrl}/webapi/student/profilePic?`
            + `courseid=${this.courseId}&studentemail=${this.studentEmail}`;
    });
```

Reply:
https://stackoverflow.com/a/45939161 Here you go :)

Reply:
I tried this link 2 days back and it failed, turns out that I had to replace `params` with `queryParams` as stated [here](https://luisreinoso.github.io/angular/2019/03/08/how-to-mock-activated-route-in-angular.html)  zzzzz🤣.
but thanks for the help! Do you need me to squash commits or is it auto squash merge

Reply:
👍 No worry. It will be auto-squashed.